src/HotChocolate/Core/src/Types/Configuration/TypeTrimmer.cs

```diff
@@ -39,6 +39,7 @@ public IReadOnlyCollection<TypeSystemObjectBase> Trim()
                 directiveType.Name.Equals(WellKnownDirectives.Deprecated))
             {
                 _touched.Add(directiveType);
+                VisitDirective(directiveType);
```

Comment:
can you add a test that shows the issue that this solves?

Reply:
@michaelstaib, it was exactly my question how to write tests in a better way as a current approach for TypeTrimmer tests uses SchemaBuilder which doesn't print scalars, but I think I figured out how to deal with that

Reply:
In GraphQL built-in scalars are not printed. You can create a dummy scalar to test that.

Reply:
But what is the error that happens?

Reply:
Is the issue that we do not visit the query directives and thus remove a scalar that is important for the directive?

Reply:
So, the trimmer has two inputs: discovered types and root types (tbh, I don't have any clue what discovered types mean here). When we iterate over root types we're using visitors which visit all directives and as a result visit all arguments. Iterating over directives from discovered types doesn't visit directives to analyze arguments and as a result, it can be the issue that one of the directives uses scalar or any other type which will not be in the final schema. In the original issue #4712 it was with `@stream` direcitive

Reply:
@michaelstaib, I've added the test which validates my fix. W/o changes, the UUID scalar will not be a part of the final schema.